app/models/fiber_channel_address.rb

```diff
@@ -0,0 +1,13 @@
+class FiberChannelAddress < ApplicationRecord
```

Comment:
Instead of manually doing a polymorphic relation, have all these tables like FiberChannelAddress and IscsiAddress inherit from `Address` itself.
Then you don't also need `self.class_by_ems`, `:adddresses`, `acts_as_miq_taggable`.
But you will need to override the getter method to return the apprioriate field.